Game/Src/Powers/Powers.gd

```diff
@@ -0,0 +1,38 @@
+extends Node
+
+onready var lightning = $LightningPower
+onready var potion = $PotionPower
+onready var slippery = $SlipperyHandsPower
+
+var rng
+
+signal glory_increment(value)
+signal health_increase(value)
+
+func _ready() -> void:
+	randomize()
+	rng = RandomNumberGenerator.new()
+
+func call_pwr(slots,pwrList,toggle,otherArgs: Dictionary = {'stats':null,'enemy':null}):
+	if not toggle.isEmpty:
+		match slots:
+			pwrList.lightning:
+				lightning.lightning(otherArgs['enemy'])
+				toggle.change_to_empty_texture()
+			pwrList.potion:
+				potion.potion(otherArgs['stats'])
+				toggle.change_to_empty_texture()
+			pwrList.slippery_hands:
+				slippery.slippery_hands()
+				toggle.change_to_empty_texture()
+
+func fill_empty(toggles,slots) -> void:
```

Comment:
I am not sure what the toggles do but can't we have an enum where 0 is empty and we check the value of the power slot rather than passing a toggle to the function?
I would also just pass the slot I want to fill to the function instead of all the slots. This way, we can take care of the logic handling the filling of power slots in the main scene

Reply:
I still use toggles because the filling animation is a property of the buttons